src/osd/PG.cc

```diff
@@ -2003,8 +2003,8 @@ void PG::activate(ObjectStore::Transaction& t,
 	pi.last_interval_started = info.last_interval_started;
 	pi.history = info.history;
 	pi.hit_set = info.hit_set;
-        // Save num_bytes for reservation request
-        peer_bytes[peer] = pi.stats.stats.sum.num_bytes;
+        // Save num_bytes for reservation request, can't be negative
+        peer_bytes[peer] = std::max<int64_t>(0, pi.stats.stats.sum.num_bytes);
```

Comment:
So the root cause is that peer's num_bytes is not quite reliable (can be totally wrong)?

Reply:
@xiexingguo There are a couple of possible explanations of how num_bytes can be off.  I think that if a roll forward or log replay could delete something twice, maybe there would be an undercount.  Or maybe something as simple as a corruption.  The num_bytes for a PG can always be fixed later with a pg repair.